pkg/controller/storagecluster/storagecluster.go

```diff
@@ -733,6 +733,25 @@ func (c *Controller) miscCleanUp(cluster *corev1.StorageCluster) error {
 	return nil
 }
 
+func preflightShouldRun(cluster *corev1.StorageCluster) bool {
+	pxVer30, _ := version.NewVersion("3.0")
+	pxVer31, _ := version.NewVersion("3.1")
+
+	// Preflight should only run freshInstall and if the PX version is 3.0.0 and above
+	if pxutil.IsFreshInstall(cluster) {
+		if pxutil.GetPortworxVersion(cluster).GreaterThanOrEqual(pxVer30) {
```

Comment:
nit: Can we extract the value of GetPortworxVersion into a variable instead of calling it twice?